client/core/trade.go

```diff
@@ -358,20 +358,14 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 			if trade.Filled() == trade.Quantity {
 				t.metaData.Status = order.OrderStatusExecuted
 			}
-		case order.CancelOrderType:
-			t.metaData.Status = order.OrderStatusCanceled
 		case order.MarketOrderType:
 			t.metaData.Status = order.OrderStatusExecuted
 		}
 	}
 	corder, cancelOrder := t.coreOrderInternal()
+	hasUnswapped := t.hasUnswappedMatches()
```

Comment:
Just above when setting `t.metaData.Status`, I think it neglects immediate TiF limits, which should bet set to `order.OrderStatusExecuted` regardless of filled amount.  https://github.com/decred/dcrdex/pull/525/files#diff-ca928c370ec8b724badec3db0dbd7bc4R356-R360.